Did the GroupBy ActionFieldGuid change recently?  (Originally 3C417469-CB89-4C22-8B5B-4F526BA7ECAA) Or does that value not exist for non-initialized workspaces?


--- Errors that can occur when trying to query DG Audits from a server ---
-- Factors we can detect and reroute --
Workspace no longer exists
 = (takes time to timeout, but can detect)
SQL connection timeout
 = (retry?)
DataGrid Audit not initialized for Workspace
 = (needs to be detected and handled at DataGridSettings layer)
Workspace is not finished upgrading 
 = (currently don't detect this, DGAudit returns with this error)

-- Beyond our control --
DataGrid Audit service errors beyond our comprehension/control
 = (Panic dump these errors?  Where can we give the greatest assistance and transparency?)
Kepler issues within the environment
 = (Panic dump these errors?  Where can we give the greatest assistance and transparency?)

Concurrency of Kepler API calls
 = (Didn't seem to have an issue before, though our errorhandling seems subpar) 
 
- Other notes -
Appears to be calling 4 times per workspace, for one userExperienceService.BuildUserExperienceModel(server, hour)
Anytime we use the abstracted 'serverAuditService', that guarentees a call for each workspace to determine what repo to use.  Can we cache this in the repo?

=================================================================
=== Found errors when trying to build up UserExperience Model ===
=================================================================
WorkspaceService: Could not connect to workspace database EDDS1123221. System.Data.SqlClient.SqlException (0x80131904): Cannot open database "EDDS1123221" requested by the login. The login failed.
Login failed for user 'eddsdbo'.
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1026213,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.997","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1046944, 1046945, 1046952, 1046964, 1046968, 1046969, 1046970, 1046978, 1046979, 1046980, 1046981, 1046982, 1046983])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1044930,"GroupBy":{"ArtifactID":1044959,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Workspace 1026213 is not finished upgrading.
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: WorkspaceId 1029199 - Mismatch of action/reponse count: Actions - 13, FilteredResponse Actions - 0, ResponseTotal - 0, MissingActions - View | Update | Update - Mass Edit | Update - Mass Replace | Update - Propagation | Document Query | Query | Import | Export | ReportQuery | RelativityScriptExecution | Pivot Query | Update - Import, {  }
   at kCura.PDB.Service.Audits.DataGridSettingsService.<ReadActionChoiceIds>d__3.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service\Audits\DataGridSettingsService.cs:line 55
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1039646,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.997","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1050890, 1050891, 1050898, 1050910, 1050914, 1050915, 1050916, 1050924, 1050925, 1050926, 1050927, 1050928, 1050929])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1044930,"GroupBy":{"ArtifactID":1044959,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1089176,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.997","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1170169,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.997","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: Field not found in workspace 1293365 for action field guild 3c417469-cb89-4c22-8b5b-4f526ba7ecaa
   at kCura.PDB.Service.Audits.DataGridSettingsService.<ReadActionChoiceIds>d__3.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service\Audits\DataGridSettingsService.cs:line 35
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: Field not found in workspace 1293514 for action field guild 3c417469-cb89-4c22-8b5b-4f526ba7ecaa
   at kCura.PDB.Service.Audits.DataGridSettingsService.<ReadActionChoiceIds>d__3.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service\Audits\DataGridSettingsService.cs:line 35
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1039646,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1050890, 1050891, 1050898, 1050910, 1050914, 1050915, 1050916, 1050924, 1050925, 1050926, 1050927, 1050928, 1050929])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1044930,"GroupBy":{"ArtifactID":1044959,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1039646,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1050890, 1050891, 1050898, 1050910, 1050914, 1050915, 1050916, 1050924, 1050925, 1050926, 1050927, 1050928, 1050929])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1044930,"GroupBy":{"ArtifactID":1044959,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1039646,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1050890, 1050891, 1050898, 1050910, 1050914, 1050915, 1050916, 1050924, 1050925, 1050926, 1050927, 1050928, 1050929])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1044930,"GroupBy":{"ArtifactID":1044959,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1089176,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1089176,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1089176,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1170169,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1170169,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
WorkspaceAuditServiceFactory,Audit: Failed to call DataGrid with error: System.Exception: DataGridService call failed for AuditQuery params: {"WorkspaceId":1170169,"StartTime":"2018-07-25T16:00:00","EndTime":"2018-07-25T16:59:59.999","ActionTypes":[1,3,4,5,6,28,29,32,33,34,35,37,47]} -- Additional details: Failed to read TotalAudits, PivotSettings: {"ObjectSetQuery":{"Condition":"(('Action' IN CHOICE [1038275, 1038276, 1038283, 1038295, 1038299, 1038300, 1038301, 1038309, 1038310, 1038311, 1038312, 1038313, 1038314])) AND (('Timestamp' >= 2018-07-25T16:00:00.00Z)) AND (('Timestamp' <= 2018-07-25T16:59:59.99Z))","Sorts":[],"RelationalField":{"ArtifactID":0,"ViewFieldID":0,"Guids":[],"Name":null},"RowCondition":null,"SearchProviderCondition":null,"ExecutingSavedSearchId":null,"SampleParameters":null,"IncludeIdWindow":null,"RankSortOrder":null,"QueryHint":null},"ArtifactTypeID":1038013,"GroupBy":{"ArtifactID":1038042,"ViewFieldID":0,"Guids":[],"Name":null},"PivotOn":null,"GroupByDateGrouping":0,"PivotOnDateGrouping":0,"MaximumNumberOfColumns":0,"MaximumNumberOfRows":0,"Timeout":0,"RawDataOnly":false,"TimeZone":null} ---> System.Exception: Read TotalAudits returned unsuccessful with message: Object reference not set to an instance of an object.
   at kCura.PDB.Service.DataGrid.DataGridService.<ReadTotalAuditsForHourAsync>d__6.MoveNext() in D:\Git_Repos\Milyli\performancedashboard\Source\kCura.PDB.Service.DataGrid\DataGridService.cs:line 117
---
